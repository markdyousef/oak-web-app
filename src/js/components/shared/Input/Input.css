:root {
    --primary-color: #5ed892;
    --main-black-color: #131517;
    --light-black-color: #2d3438;
    --dark-gray-color: #737a80;
    --main-gray-color: #f2f2f2;
    --light-gray-color: #fafafa;
    --main-white-color: #fff;
    --line-gray-color: #e5e5e5;
}

.container {
    width: 100%;
    position: relative;
    max-width: 400px;
}

.container input {
    font-size: 16px;
    padding: 10px;
    display: block;
    width: 100%;
    font-weight: lighter;
    background-color: rgba(255, 0, 0, 0.0);
    border: 1px solid var(--line-gray-color);
    height: 50px;
    border-radius: 3px;
}
