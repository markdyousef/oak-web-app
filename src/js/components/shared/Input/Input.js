// @flow
import React, { PropTypes } from 'react';
import styled from 'styled-components';
import colors from '../../../styles/colors';

const valid = (notValid) => {
    if (notValid) {
        return `
        background-color: rgba(232, 115, 133, 0.1);
        border-color: rgba(232, 115, 133, 0.75)
        `;
    }
    return `
    background-color: #fff;
    &:focus {

        border-color: #67C198
    }
    &:hover {
        border-color: #B5D3C9
    }`;
};

const H5 = styled.h5`
    font-size: 14px;
    font-weight: lighter;
    color: ${colors.darkGrey};
    margin-bottom: 5px;
`;

const Default = styled.input`
    padding: 12px 8px;
    width: 100%;
    max-width: 400px;
    border-radius: 3px;
    border: 1px solid ${colors.lightGrey};
    font-size: 15px;
    border-radius: 3px;
    outline: none;
    ${props => valid(props.notValid)}
`;

const Input = ({ ...props }:Object) => {
    return (
        <div>
            <H5>{props.title}</H5>
            <Default
                type={props.type || 'text'}
                value={props.value}
                placeholder={props.placeholder}
                onChange={event => props.onChange(event.target.value)}
                notValid={props.notValid}
                onKeyDown={props.onKeyDown}
            />
        </div>
    );
};

Input.propTypes = {
    title: PropTypes.string,
    value: PropTypes.string,
    placeholder: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    type: PropTypes.string,
    notValid: PropTypes.bool,
    onKeyDown: PropTypes.func
};

Input.defaultProps = {
    title: null,
    value: null,
    placeholder: null,
    type: null,
    notValid: false,
    onKeyDown: null
};

export default Input;
