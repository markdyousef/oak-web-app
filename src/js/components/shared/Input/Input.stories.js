import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import Input from './Input';

storiesOf('Input', module)
    .add('Default Input', () => (
        <Input
            onClick={action('Fuck Yeah!')}
            text="Cancel"
            placeholder="Name"
        />
    ))
    .add('Default Input notValid', () => (
        <Input
            onClick={action('Fuck Yeah!')}
            text="Cancel"
            placeholder="Name"
            notValid
        />
    ));
