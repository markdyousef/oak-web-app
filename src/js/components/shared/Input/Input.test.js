import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

it('renders correctly', () => {
    const props = {
        title: 'Name',
        onChange: () => {}
    };
    const component = renderer.create(
        <Input {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
