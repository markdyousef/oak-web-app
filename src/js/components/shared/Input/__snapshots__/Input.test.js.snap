exports[`test renders correctly 1`] = `
<div>
  <h5
    className="eKaiQM">
    Name
  </h5>
  <input
    className="llFZIf"
    onChange={[Function]}
    onKeyDown={null}
    placeholder={null}
    type="text"
    value={null} />
</div>
`;
